File: C-to-C# Port Wizard/assign3Samples/fileIn.c
/*
 * fileIn.c
 *
 * a simple source file demonstrating file input
 */

#include <stdio.h>
#include <string.h>

int main (void)
{
	// local variables
	int	fileLine = 1;
	FILE 	*fp;
	char	inBuff[100];

	// open the file
	fp = fopen ("test.txt", "r");
	if (fp == NULL) 
	{
		printf ("**Error: Unable to open file : test.txt for reading\n");
		return 1;
	}

	/* Walk through the file line by line counting and displaying
	   the file contents */
	printf("The following are the lines read from file \"test.txt\"\n");
	while (fgets(inBuff, 100, fp) != NULL)
	{
		printf ("%2d) [Length : %3d] %s", fileLine, strlen(inBuff), inBuff);
		fileLine++;
	}
	printf("<<< EOF\n");

	fclose (fp);

	return 0;
}

File: C-to-C# Port Wizard/PortWizard/PortWizard/bin/Debug/assign3Samples/fileIn.cs
/*
* fileIn.c
*
* a simple source file demonstrating file input
*/

using System;
using System.IO;
using System.Text;


namespace Something
{
	class Program
	{
		static void Main()
		{
			// local variables
			int	fileLine = 1;
			FileStream fp;
			char[] inBuff = new char[100];
			
			// open the file
			fp = File.Open ("test.txt", FileMode.OpenOrCreate, FileAccess.Read);
			if (fp == null) 
			{
				Console.WriteLine ("**Error: Unable to open file : test.txt for reading");
				return;
			}
			
			/* Walk through the file line by line counting and displaying
			the file contents */
			Console.WriteLine("The following are the lines read from file \"test.txt\"");
			while ( fp.Read(Encoding.ASCII.GetBytes(inBuff, 0, inBuff.Length), 0,  100) != 0)
			{
			Console.Write ("{0}) [Length : {1}] {2}", fileLine, inBuff.Length, inBuff);
				fileLine++;
			}
			Console.WriteLine("<<< EOF");
			
			fp.Close();
			
			return;
		}

	}
}
